### PR TITLE
test: Pin and document TEST_DIR_PATH_ELEMENT, SeedRand::FIXED_SEED

### PR DESCRIPTION
**Pull request by maflcko**
Two small test changes:

* A refactor to update the name and documentation around `SeedRand::FIXED_SEED`.
* A change to extract and document `TEST_DIR_PATH_ELEMENT`, and to change its value to better match the `TMPDIR_PREFIX` in functional tests. The value previously included `PACKAGE_NAME`, which is cute, but doesn't explain why it was used (to include a space). So just use `test_common bitcoin` to achieve the same with less effort.

### CODE REVIEW

**Review by hodlinator — COMMENTED** [review]
Concept ACK fa9c3aecb18e68cfaeae44aa0daffec020ca3927

Not entirely sure why the two changes are combined in one PR but they are small enough and both touch **setup_common.cpp**.

Good to see `SeedRandomForTest` no longer defaults which enum value to use (it was explicitly passed everywhere anyway). :+1: 

No remaining references to `PACKAGE_NAME` in tests, nor "test_common_Bitcoin Core" in project. Shortening to "test_common bitcoin" seems like it should be tolerable for forks.

File: src/test/util/setup_common.cpp [hunk 1]
```diff
@@ -77,6 +75,7 @@ const std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
 
 /** Random context to get unique temp data dirs. Separate from m_rng, which can be seeded from a const env var */
 static FastRandomContext g_rng_temp_path;
+constexpr inline auto TEST_DIR_PATH_ELEMENT{"test_common bitcoin"}; // Includes a space to catch possible path escape issues.
```
[the remark this answers is not in the document]
Comment (hodlinator) [reply src/test/util/setup_common.cpp:78]: Okay, would have at least tweaked the comment a bit in that case. But I'll chalk this down to me still getting familiar with this project's style of working with the code.

[the remark this answers is not in the document]
Comment (hodlinator) [reply src/test/util/setup_common.cpp:78]: I agree it's self-explanatory, and it is closely related, but the comment does not apply to it. How about putting it first, still grouping them together?
```C++
constexpr inline auto TEST_DIR_PATH_ELEMENT{"test_common bitcoin"}; // Includes a space to catch possible path escape issues.
/** Random context to get unique temp data dirs. Separate from m_rng, which can be seeded from a const env var */
static FastRandomContext g_rng_temp_path;
```

**Review by hodlinator — APPROVED** [review]
ACK fa84f9decd224ea1c25dc7095bad70a48fa1a534

`git range-diff master fa9c3aec fa84f9de`

Only one minor change (that I suggested) since [prior review](https://github.com/bitcoin/bitcoin/pull/30748#pullrequestreview-2268619255).

Passed `ctest -j <cores> --test-dir build`.

**Review by ryanofsky — APPROVED** [review]
Code review ACK fa84f9decd224ea1c25dc7095bad70a48fa1a534

IMO, `TEST_DIR_PATH_ELEMENT` would be clearer as `TESTS_DIR_NAME` and `"test_common bitcoin"` would be clearer as `"bitcoin tests"` but current names seem fine too.

**Pull request merged**